File: POO_2026_1/src/MeuPacote/Vetor.java
package MeuPacote;

import java.util.Scanner;

public class Vetor {
  
  public static void entrada(String nome, int x[]) {
    Scanner ler = new Scanner(System.in);
    
    int i, n = x.length;
    
    for(i=0; i<n; i++) {
      System.out.printf("%do. item de %d, %s[%d] = ", (i+1), n, nome, i);
      x[i] = ler.nextInt();
    }
    
    System.out.println();
  }
  
  public static String mostrar(String nome, int x[]) {
    int i, n = x.length;
    String result;
    
    result = nome + " = { ";
    for(i=0; i<n; i++) {
      result += String.format("%d ", x[i]);
    }
    result += "}";
    
    return(result);    
  }
  
  public static int sum(int x[]) {
    int i, n = x.length, sm = 0;
    
    for(i=0; i<n; i++) {
      sm = sm + x[i];
    }
    
    return(sm);
  }
  
  public static double avg(int x[]) {
    return(sum(x) / (double)x.length);
  }
  
  public static int min(int x[]) {
    int i, n = x.length, menor = x[0];
    for(i=1; i<n; i++) {
      if (x[i] < menor)
          menor = x[i];
    }
    return(menor);
  }
  
  public static int max(int x[]) {
    int i, n = x.length, maior = x[0];
    for(i=1; i<n; i++) {
      if (x[i] > maior)
          maior = x[i];
    }
    return(maior);
  }
  
  public static String toString(String nome, int x[]) {
    String result = mostrar(nome, x) + "\n\n";
    
    result += String.format("Soma dos valores = %d\n", sum(x)) +
              String.format("Valor médio      = %.2f\n", avg(x)) +
              String.format("Menor valor      = %d\n", min(x)) +
              String.format("Maior valor      = %d\n\n", max(x));
    
    return result;
  }
  
}

File: CPP/Abr_14_Usando_MeuPacote_CPP/Usando_Vetor/Vetor.cpp
#include "Vetor.h"

void entrada(string nome, int x[], int n) {
  for(int i=0; i<n; i++) {
    cout << (i+1) << "o. item de " << n << ", " << nome << "[" << i << "] = ";
    cin >> x[i];
  }
  cout << endl;
}

string mostrar(string nome, int x[], int n) {
  string result = nome + " = { ";

  for(int i=0; i<n; i++) {
    result += to_string(x[i]) + " ";
  }
  result += "}\n";

  return(result);
}

int sum(int x[], int n) {
  int result = 0;
  for(int i=0; i<n; i++) {
    result += x[i];
  }
  return(result);
}

double avg(int x[], int n) {
  return(sum(x, n) / (double)n);
}

int min(int x[], int n) {
  int menorValor = x[0];
  for(int i=1; i<n; i++) {
    if (x[i] < menorValor) {
       menorValor = x[i];
    }
  }
  return(menorValor);
}

int max(int x[], int n) {
  int maiorValor = x[0];
  for(int i=1; i<n; i++) {
    if (x[i] > maiorValor) {
       maiorValor = x[i];
    }
  }
  return(maiorValor);
}

string toString(string nome, int x[], int n) {
  char result[200];

  sprintf(result, "%s\n"
                  "Soma dos itens        = %d\n"
                  "Valor Médio dos itens = %.2f\n"
                  "Menor Valor dos itens = %d\n"
                  "Maior Valor dos itens = %d\n",
                  mostrar(nome, x, n).c_str(),
                  sum(x, n),
                  avg(x, n),
                  min(x, n),
                  max(x, n));

  return(result);
}
